pages/dashboard.js

```diff
@@ -45,39 +45,136 @@ export default function Dashboard(props) {
 
   const [gisBenefit, setGisBenefit] = useState()
   const [gisLoaded, setGisLoaded] = useState(false)
-  const [gisError, setGisError] = useState(false)
+  const [gisError, setGisError] = useState()
 
   const [sebBenefit, setSebBenefit] = useState()
   const [sebLoaded, setSebLoaded] = useState(false)
   const [sebError, setSebError] = useState()
 
+  const [allBenefits, setAllBenefits] = useState()
+
   useEffect(() => {
-    async function fetchProgramData(program, setData, setError, setLoaded) {
-      try {
-        const res = await fetch(`/api/programData/${program}`)
-        if (res.ok) {
-          if (res.status === 200) {
-            setData(await res.json())
+    if (
+      !cppLoaded &&
+      !cppdLoaded &&
+      !eiLoaded &&
+      !oasLoaded &&
+      !gisLoaded &&
+      !sebLoaded
+    ) {
```

Comment:
I'm wondering if these would be better placed around each of their own 'fetchProgramData' ? so that each program operates a little more independently.

Reply:
I see what you mean, if you prefer that, we can have it that way. Say the word and I'll make the change.

Reply:
You're the author on this, can leave it as it. It's consistent with the other useEffect.